WalletWasabi/Helpers/PowerSaving/LinuxInhibitorTask.cs

```diff
@@ -216,19 +217,19 @@ public static LinuxInhibitorTask Create(InhibitWhat what, TimeSpan basePeriod, s
 		if (gui == GraphicalEnvironment.Gnome)
 		{
 			string inhibitArgument = ConstructInhibitArgument(what);
-			command = "gnome-session-inhibit";
+			command = "sudo gnome-session-inhibit";
 			arguments = $"--reason \"{reason}\" --inhibit {inhibitArgument} {innerCommand}";
 		}
 		else if (gui == GraphicalEnvironment.Mate)
 		{
 			string inhibitArgument = ConstructInhibitArgument(what);
-			command = $"mate-session-inhibit";
+			command = $"sudo mate-session-inhibit";
 			arguments = $"--reason \"{reason}\" --inhibit {inhibitArgument} {innerCommand}";
```

Comment:
It would have to be:

```suggestion
			command = $"sudo";
			arguments = $"mate-session-inhibit --reason \"{reason}\" --inhibit {inhibitArgument} {innerCommand}";
```

but I don't believe this is a correct to fix the issue. Basically, WW is started under a regular user and the WW process cannot run as a child process a process with elevated privileges.

Reply:
Max might try to run WW under the root account, if he's fine with that.

Reply:
did `sudo dotnet run`, it crashes with

```
2022-03-11 18:57:04.484 [8] INFO	TorProcessManager.RestartingLoopAsync (97)	Tor is already running on 127.0.0.1:37150.
2022-03-11 18:57:04.508 [8] ERROR	TorProcessManager.RestartingLoopAsync (143)	Unexpected problem in starting Tor. Exception: System.IO.FileNotFoundException: Could not find file '/root/.walletwasabi/client/control_auth_cookie'.
File name: '/root/.walletwasabi/client/control_auth_cookie'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.File.ReadAllBytes(String path)
   at WalletWasabi.Tor.TorProcessManager.InitTorControlAsync(CancellationToken token) in WalletWasabi/Tor/TorProcessManager.cs:line 249
   at WalletWasabi.Tor.TorProcessManager.RestartingLoopAsync(CancellationToken globalCancellationToken) in WalletWasabi/Tor/TorProcessManager.cs:line 98
2022-03-11 18:57:04.513 [8] INFO	Global.Dispose (237)	TorProcessManager.Start finished in 64 milliseconds.
2022-03-11 18:57:05.622 [1] DEBUG	Program.Main (81)	System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Threading.Tasks.TaskExtensions.WithAwaitCancellationAsync[T](Task`1 task, CancellationToken cancellationToken) in WalletWasabi/Extensions/TaskExtensions.cs:line 14
   at WalletWasabi.Tor.TorProcessManager.StartAsync(CancellationToken cancellationToken) in WalletWasabi/Tor/TorProcessManager.cs:line 63
   at WalletWasabi.Fluent.Global.StartTorProcessManagerAsync(CancellationToken cancel) in WalletWasabi.Fluent/Global.cs:line 240
   at WalletWasabi.Fluent.Global.InitializeNoWalletAsync(TerminateService terminateService) in WalletWasabi.Fluent/Global.cs:line 153
   at WalletWasabi.Fluent.Desktop.Program.<>c.<<BuildAvaloniaApp>b__9_1>d.MoveNext() in WalletWasabi.Fluent.Desktop/Program.cs:line 189
--- End of stack trace from previous location ---
   at WalletWasabi.Fluent.App.<OnFrameworkInitializationCompleted>b__11_0() in WalletWasabi.Fluent/App.axaml.cs:line 72
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Avalonia.Threading.AvaloniaSynchronizationContext.<>c__DisplayClass5_0.<Post>b__0() in /_/src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs:line 33
   at Avalonia.Threading.JobRunner.Job.Avalonia.Threading.JobRunner.IJob.Run() in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 181
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 37
   at Avalonia.X11.X11PlatformThreading.HandleX11(CancellationToken cancellationToken) in /_/src/Avalonia.X11/X11PlatformThreading.cs:line 169
   at Avalonia.X11.X11PlatformThreading.RunLoop(CancellationToken cancellationToken) in /_/src/Avalonia.X11/X11PlatformThreading.cs:line 245
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 65
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 120
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 209
   at WalletWasabi.Fluent.Desktop.Program.Main(String[] args) in WalletWasabi.Fluent.Desktop/Program.cs:line 75
2022-03-11 18:57:05.623 [1] DEBUG	TerminateService.Terminate (102)	Start shutting down the application.
```

Reply:
running this branch with sudo has the same error.

but running master with sudo runs without any errors.

Reply:
> but running master with sudo runs without any errors.

Yeah, this is what I meant. Honestly, it's hard to say why your environment requires sudo for inhibiting sleep. It seems it differs from distribution to distribution. AFAIK there is no good documentation explaining the behavior. 

Maybe we can ask on a Ubuntu forum and a Qubes forum, I don't know about a better solution at the moment.

Reply:
I also tried `sudo HOME=$HOME dotnet run`, but that crashes with
```
2022-03-11 20:25:07.261 [17] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:07.264 [17] INFO	Global.Dispose (237)	TorProcessManager.Start finished in 368 milliseconds.
2022-03-11 20:25:07.266 [17] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:07.266 [17] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:07.519 [19] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:07.520 [19] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:07.520 [19] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:07.779 [8] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:07.780 [8] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:07.780 [8] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:08.032 [8] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:08.033 [8] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:08.033 [8] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:08.287 [19] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:08.290 [19] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:08.290 [19] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:08.550 [8] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:08.551 [8] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
2022-03-11 20:25:08.551 [8] INFO	TorProcessManager.StartProcess (236)	Starting Tor process ...
2022-03-11 20:25:08.810 [17] ERROR	TorProcessManager.EnsureRunningAsync (195)	Tor process failed to start!
2022-03-11 20:25:08.812 [19] DEBUG	TorProcessManager.StartProcess (233)	Environment variable 'LD_LIBRARY_PATH' set to: '/home/user/WalletWasabi/WalletWasabi.Fluent.Desktop/bin/Debug/net6.0/Microservices/Binaries/lin64/Tor'.
```